Modules/Core/Common/wrapping/itkFunctionBase.wrap

```diff
@@ -28,7 +28,15 @@ itk_wrap_class("itk::FunctionBase" POINTER)
 
     # Required by BSplineInterpolationWeightFunction
     itk_wrap_template("${ITKM_CID2}${ITKM_FAD9}" "${ITKT_CID2}, ${ITKT_FAD9}")
-    itk_wrap_template("${ITKM_CID3}${ITKM_FAD64}" "${ITKT_CID3}, ${ITKT_FAD64}")
+    foreach(d ${ITK_WRAP_IMAGE_DIMS})
+      # Wrapping for spline order 3, components = (SplineOrder + 1)^SpaceDimension
+      set(comp 4)
+      math(EXPR stop "${d}-1")
+      foreach(i RANGE 1 ${stop})
+        math(EXPR comp "${comp}*4")
+      endforeach()
```

Comment:
`foreach(i RANGE 1 ${stop})` [has undefined behavior](https://cmake.org/cmake/help/latest/command/foreach.html) if `${stop}` is 0, which could happen if 1 is ever a supplied value in the `ITK_WRAP_IMAGE_DIMS` list.  It would be safer to use
```cmake
set(comp 1)
foreach(i RANGE 1 ${d})
  math(EXPR comp "${comp}*4")
endforeach()
```

Reply:
:+1: